src/domain/entities/figma-design-identifier.ts

```diff
@@ -40,12 +40,16 @@ export class FigmaDesignIdentifier {
 		);
 
 		const fileKey = pathComponents[filePathComponentId + 1];
+		const branchFileKey =
+			pathComponents[filePathComponentId + 2] === 'branch'
+				? pathComponents[filePathComponentId + 3]
+				: undefined;
 		const nodeId = url.searchParams.get('node-id')?.replace('-', ':');
 
 		if (!fileKey)
 			throw new Error(`Received invalid Figma URL: ${url.toString()}`);
 
-		return new FigmaDesignIdentifier(fileKey, nodeId);
+		return new FigmaDesignIdentifier(branchFileKey ?? fileKey, nodeId);
```

Comment:
**question:** `fileKey` is currently used to call several Figma APIs:

- `GET /v1/files/:fileKey/meta` for fetching _Design_ metadata.
- `GET `/v1/files/:fileKey` for fetching _Design_ metadata.

- `GET /v1/files/:fileKey/dev_resources` for fetching _Dev Resources_ for a _Design_.
- `POST `/v1/dev_resources` for created a _Dev Resource_ for a _Design_.
- `DELETE /v1/files/:fileKey/dev_resources/:devResourceId` for deleting a _Dev Resources_ for a _Design_.

- `POST /v2/webhooks` for creating webhooks for a _Team_ (`fileKey` is not used directly but there we still need to receive events for the design).

Do we expect `branchFileKey` to be fully interchangeable with `fileKey`, so we get the same behavior for branch-level designs?

Reply:
yep! they're interchangeable when interfacing with the REST API. Created branches also get added to the original file's team so they should still receive webhooks